File: _ConvertedToCSharp/Library/ProcessedFileEventArgs.cs
//=================================
//Deniz Esen
//Econ Dimension GmbH
//http://www.econdimension.com
//d.esen@econdimension.com
//=================================
// You are free to use, change and
// redistribute this Class or parts
// of it.
// No Support.
//=================================
/// <summary>
/// Contains informations about the conversion of a file.
/// </summary>
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Windows.Forms;
namespace Econ.NetVert
{
  public class ProcessedFileEventArgs : EventArgs
  {
    public enum ConverterFileTypes
    {
      CodeFile,
      VSProjectFile,
      ASPXFile,
      OtherFile
    }

    public enum ConverterOperations
    {
      Before_Conversion = 0,
      Success_Creating = 10,
      Success_Overwriting = 11,
      FlatCopy = 15,
      Skipped_Exists = 20,
      Skipped_Error = 21
    }

    //DECLARATIONS
    private ConverterFileTypes FFileType;
    private string FInputFile;
    private string FInputSource;
    private string FOutputFile;
    private string FOutputSource;
    private ConverterOperations FOperation;
    private ConverterLanguagesAutodetectable FConverterLanguage;
    private string FConverterError;

    //CONSTRUCTOR
    internal ProcessedFileEventArgs(ConverterFileTypes fileType, string inFile, string inSource, string outFile, string outSource, ConverterOperations convOperation, ConverterLanguagesAutodetectable convLang, string convError)
    {
      FFileType = fileType;
      FInputFile = inFile;
      FInputSource = inSource;
      FOutputFile = outFile;
      FOutputSource = outSource;
      FOperation = convOperation;
      FConverterLanguage = convLang;
      FConverterError = convError;
    }

    //PUBLIC
#region PUBLIC Properties

    public ConverterFileTypes FileType {
      get { return FFileType; }
    }

    public string InputFile {
      get { return FInputFile; }
    }

    public string InputSource {
      get { return FInputSource; }
      set { FInputSource = value; }
    }

    public Int32 InputLinesOfCode {
      get { return Split(FInputSource, vbCrLf).Length; }
    }

    public string OutputFile {
      get { return FOutputFile; }
      set { FOutputFile = value; }
    }

    public string OutputSource {
      get { return FOutputSource; }
      set { FOutputSource = value; }
    }

    public Int32 OutputLinesOfCode {
      get { return Split(FOutputSource, vbCrLf).Length; }
    }

    public ConverterOperations ConverterOperation {
      get { return FOperation; }
    }

    public ConverterLanguagesAutodetectable ConverterLanguage {
      get { return FConverterLanguage; }
    }

    public string ConverterError {
      get { return FConverterError; }
    }

#endregion

    public override string ToString()
    {
      return this.ToString("", "");
    }

    public string ToString(string inputRootPath, string outputRootPath)
    {
      string TmpFT = "";
      string TmpOutLng = "";
      string RetS = "";

      switch (FFileType) {
        case ConverterFileTypes.CodeFile:
          TmpFT = "Codefile";
        case ConverterFileTypes.ASPXFile:
          TmpFT = "ASP.NET File";
          switch (FConverterLanguage) {
            case ConverterLanguagesAutodetectable.VBNetToCSharp:
              TmpOutLng = " [C#]";
            case ConverterLanguagesAutodetectable.CSharpToVBNet:
              TmpOutLng = " [VB.NET]";
          }
        case ConverterFileTypes.VSProjectFile:
          TmpFT = "Visual Studio Project";
        case ConverterFileTypes.OtherFile:
          TmpFT = "File";
      }
      switch (FOperation) {
        case ConverterOperations.Before_Conversion:
          RetS = "Reading " + TmpFT + " '" + TrimmPath(FInputFile, inputRootPath) + "' (" + InputLinesOfCode.ToString + " lines, " + InputSource.Length.ToString + " bytes)";
        case ConverterOperations.Skipped_Error:
          RetS = "Error in " + TmpFT + " '" + TrimmPath(FInputFile, inputRootPath) + "'" + vbCrLf + FConverterError;
        case ConverterOperations.Skipped_Exists:
          RetS = "Skipped, " + TmpFT + " exists '" + TrimmPath(FOutputFile, outputRootPath) + "'";
        case ConverterOperations.Success_Creating:
          RetS = "Creating " + TmpFT + TmpOutLng + " '" + TrimmPath(FOutputFile, outputRootPath) + "' (" + OutputLinesOfCode.ToString + " lines, " + OutputSource.Length.ToString + " bytes)";
        case ConverterOperations.Success_Overwriting:
          RetS = "Overwriting " + TmpFT + TmpOutLng + " '" + TrimmPath(FOutputFile, outputRootPath) + "' (" + OutputLinesOfCode.ToString + " lines, " + OutputSource.Length.ToString + " bytes)";
        case ConverterOperations.FlatCopy:
          RetS = "Copy " + TmpFT + " '" + TrimmPath(FOutputFile, outputRootPath) + "'";
      }
      return RetS;
    }

    //FRIEND
    internal void ModifyStatus(ConverterOperations convStatus, string convError = "", ConverterLanguagesAutodetectable convLang = ConverterLanguagesAutodetectable.Autodetect)
    {
      FOperation = convStatus;
      FConverterError = convError;
      if (convLang != ConverterLanguagesAutodetectable.Autodetect) {
        FConverterLanguage = convLang;
      }
    }

    //PRIVATE
    private string TrimmPath(string fullPath, string removePath)
    {
      string RetS;

      if ((removePath != "") && (fullPath.Length > removePath.Length) && (fullPath.Substring(0, removePath.Length).ToLower == removePath.ToLower)) {
        RetS = fullPath.Substring(removePath.Length, fullPath.Length - removePath.Length).TrimStart('\\');
      } else {
        RetS = fullPath;
      }
      return RetS;
    }

  }
}

File: Deployment_Source/Library/ProcessedFileEventArgs.vb
'=================================
'Deniz Esen
'Econ Dimension GmbH
'http://www.econdimension.com
'd.esen@econdimension.com
'=================================
' You are free to use, change and
' redistribute this class or parts
' of it.
' No Support.
'=================================

''' <summary>
''' Contains informations about the conversion of a file.
''' </summary>
Public Class ProcessedFileEventArgs
  Inherits EventArgs

  Public Enum ConverterFileTypes
    CodeFile
    VSProjectFile
    ASPXFile
    OtherFile
  End Enum

  Public Enum ConverterOperations
    Before_Conversion = 0
    Success_Creating = 10
    Success_Overwriting = 11
    FlatCopy = 15
    Skipped_Exists = 20
    Skipped_Error = 21
    'Common_Error = 99
  End Enum

  'DECLARATIONS
  Private FFileType As ConverterFileTypes
  Private FInputFile As String
  Private FInputSource As String
  Private FOutputFile As String
  Private FOutputSource As String
  Private FOperation As ConverterOperations
  Private FConverterLanguage As ConverterLanguagesAutodetectable
  Private FConverterError As String

  'CONSTRUCTOR
  Friend Sub New(ByVal fileType As ConverterFileTypes, _
                 ByVal inFile As String, _
                 ByVal inSource As String, _
                 ByVal outFile As String, _
                 ByVal outSource As String, _
                 ByVal convOperation As ConverterOperations, _
                 ByVal convLang As ConverterLanguagesAutodetectable, _
                 ByVal convError As String)
    FFileType = fileType
    FInputFile = inFile
    FInputSource = inSource
    FOutputFile = outFile
    FOutputSource = outSource
    FOperation = convOperation
    FConverterLanguage = convLang
    FConverterError = convError
  End Sub

  'PUBLIC
#Region "PUBLIC Properties"

  Public ReadOnly Property FileType() As ConverterFileTypes
    Get
      Return FFileType
    End Get
  End Property

  Public ReadOnly Property InputFile() As String
    Get
      Return FInputFile
    End Get
  End Property

  Public Property InputSource() As String
    Get
      Return FInputSource
    End Get
    Set(ByVal value As String)
      FInputSource = value
    End Set
  End Property

  Public ReadOnly Property InputLinesOfCode() As Int32
    Get
      Return Split(FInputSource, vbCrLf).Length
    End Get
  End Property

  Public Property OutputFile() As String
    Get
      Return FOutputFile
    End Get
    Set(ByVal value As String)
      FOutputFile = value
    End Set
  End Property

  Public Property OutputSource() As String
    Get
      Return FOutputSource
    End Get
    Set(ByVal value As String)
      FOutputSource = value
    End Set
  End Property

  Public ReadOnly Property OutputLinesOfCode() As Int32
    Get
      Return Split(FOutputSource, vbCrLf).Length
    End Get
  End Property

  Public ReadOnly Property ConverterOperation() As ConverterOperations
    Get
      Return FOperation
    End Get
  End Property

  Public ReadOnly Property ConverterLanguage() As ConverterLanguagesAutodetectable
    Get
      Return FConverterLanguage
    End Get
  End Property

  Public ReadOnly Property ConverterError() As String
    Get
      Return FConverterError
    End Get
  End Property

#End Region

  Public Overloads Overrides Function ToString() As String
    Return Me.ToString("", "")
  End Function

  Public Overloads Function ToString(ByVal inputRootPath As String, _
                                     ByVal outputRootPath As String) As String
    Dim TmpFT As String = ""
    Dim TmpOutLng As String = ""
    Dim RetS As String = ""

    Select Case FFileType
      Case ConverterFileTypes.CodeFile
        TmpFT = "Codefile"
      Case ConverterFileTypes.ASPXFile
        TmpFT = "ASP.NET File"
        Select Case FConverterLanguage
          Case ConverterLanguagesAutodetectable.VBNetToCSharp
            TmpOutLng = " [C#]"
          Case ConverterLanguagesAutodetectable.CSharpToVBNet
            TmpOutLng = " [VB.NET]"
        End Select
      Case ConverterFileTypes.VSProjectFile
        TmpFT = "Visual Studio Project"
      Case ConverterFileTypes.OtherFile
        TmpFT = "File"
    End Select
    Select Case FOperation
      Case ConverterOperations.Before_Conversion
        RetS = "Reading " & TmpFT & " '" & TrimmPath(FInputFile, inputRootPath) & "' (" & InputLinesOfCode.ToString & " lines, " & InputSource.Length.ToString & " bytes)"
      Case ConverterOperations.Skipped_Error
        RetS = "Error in " & TmpFT & " '" & TrimmPath(FInputFile, inputRootPath) & "'" & vbCrLf & FConverterError
      Case ConverterOperations.Skipped_Exists
        RetS = "Skipped, " & TmpFT & " exists '" & TrimmPath(FOutputFile, outputRootPath) & "'"
      Case ConverterOperations.Success_Creating
        RetS = "Creating " & TmpFT & TmpOutLng & " '" & TrimmPath(FOutputFile, outputRootPath) & "' (" & OutputLinesOfCode.ToString & " lines, " & OutputSource.Length.ToString & " bytes)"
      Case ConverterOperations.Success_Overwriting
        RetS = "Overwriting " & TmpFT & TmpOutLng & " '" & TrimmPath(FOutputFile, outputRootPath) & "' (" & OutputLinesOfCode.ToString & " lines, " & OutputSource.Length.ToString & " bytes)"
      Case ConverterOperations.FlatCopy
        RetS = "Copy " & TmpFT & " '" & TrimmPath(FOutputFile, outputRootPath) & "'"
    End Select
    Return RetS
  End Function

  'FRIEND
  Friend Sub ModifyStatus(ByVal convStatus As ConverterOperations, _
                          Optional ByVal convError As String = "", _
                          Optional ByVal convLang As ConverterLanguagesAutodetectable = ConverterLanguagesAutodetectable.Autodetect)
    FOperation = convStatus
    FConverterError = convError
    If convLang <> ConverterLanguagesAutodetectable.Autodetect Then
      FConverterLanguage = convLang
    End If
  End Sub

  'PRIVATE
  Private Function TrimmPath(ByVal fullPath As String, _
                             ByVal removePath As String) As String
    Dim RetS As String

    If (removePath <> "") AndAlso _
       (fullPath.Length > removePath.Length) AndAlso _
       (fullPath.Substring(0, removePath.Length).ToLower = removePath.ToLower) Then
      RetS = fullPath.Substring(removePath.Length, fullPath.Length - removePath.Length).TrimStart("\"c)
    Else
      RetS = fullPath
    End If
    Return RetS
  End Function

End Class
